### ISSUE TITLE
support `network_mirror` configuration of `.terraformrc`

### ISSUE THREAD

**Issue by daniel-ciaglia** [boring-registry/boring-registry#142]
I implemented a rather rudimentary mirror script to get versions of used providers and their signatures and store them in `/providers` directory of the boring registry. Sadly this doesn't match the expectations of terraform. (the needed files are present)

The key difference is the `<namespace>/<provider>`, which gets extended by terraform by `registry.terraform.io`

`.terraformrc`  is
```hcl
provider_installation {
  network_mirror {
    url = "https://terraform-registry.<client>/"
    include = ["*/*"]
  }
}
```

```
Error: Failed to query available provider packages

│ Could not retrieve the list of available versions for provider hashicorp/random: provider registry.terraform.io/hashicorp/random was not found in any of the search locations
│
│   - provider mirror at https://terraform-registry.<client>/
```

```shell
❯ curl -H "Authorization: Bearer xxx" https://terraform-registry.<client>/v1/providers/registry.terraform.io/hashicorp/aws/versions
404 page not found
❯ curl -H "Authorization: Bearer xxx" https://terraform-registry.<client>/v1/providers/hashicorp/aws/versions
{"versions":[{"version":"5.15.0","platforms":[{"os":"darwin","arch":"arm64"},{"os":"linux","arch":"amd64"}]}]}
```

see: https://github.com/TierMobility/boring-registry/issues/21

## filesystem layout `/providers`
```
.
├── hashicorp
│   ├── archive
│   ├── aws
│   ├── cloudinit
│   ├── helm
│   ├── kubernetes
│   ├── null
│   ├── random
│   ├── time
│   └── tls
└── registry.terraform.io
    └── hashicorp
        ├── archive
        ├── aws
        ├── cloudinit
        ├── helm
        ├── kubernetes
        ├── null
        ├── random
        ├── time
        └── tls
```

**Comment by daniel-ciaglia** [comment]
Using the beta build from PR#143 I can confirm the feature is working. Waiting for the 0.12.0 release to close the issue

**Comment by oliviermichaelis** [comment]
Closing this issue as I released `v0.12.0` yesterday :)
